Fix test deprecations

benjaminwols opened the pull request:
Minitest has new syntax which showed a lot of deprecation warnings in the test suite.
This PR converts to the new syntax.

mvz commented on Rakefile:
```diff
@@ -4,6 +4,7 @@ require "rake/testtask"
 Rake::TestTask.new do |t|
   t.libs.push 'spec'
   t.test_files = FileList['spec/**/*_spec.rb']
+  t.warning = false
```
Any particular reason to turn of warnings here?

benjaminwols replied:
No particular reason, I can revert the change if you want?

mvz replied:
I prefer keeping warnings on, but I'll let @maartenvanvliet have the final say.

benjaminwols replied:
I've reverted the change and merged newest master so tests are run

maartenvanvliet replied:
Thx, yes I prefer keeping the warnings

benjaminwols replied:
done

benjaminwols commented:
@maartenvanvliet I've updated the Readme badge to Github Actions and removed the deprecated code climate gem and updated Github actions to push coverage to code climate

maartenvanvliet reviewed (changes requested):
Can't say I'm a fan of the `_(...).must_equal` syntax, and would probably prefer plain `assert_equal` assertions over this. But the library already uses specs and this is the new way so 🤷

Thanks for fixing it all up!
